internal/devtools/source.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package devtools
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+var SourceDir = sourceDir
+
+func sourceDir() (string, error) {
```

Comment:
Why bother with this setup, just make the function public?

Reply:
I think I had one test that was using patching this (in the commands package I think), but I'll look to change that to avoid the patching, because that's ugly.